spec/features/pets/delete_logic_spec.rb

```diff
@@ -0,0 +1,104 @@
+        status: true)
+
+      @application_1 = Apply.create!(name: "Third Person",
+        address: "96 There St",
+        city: "CityPlace",
+        state: "StateLocation",
+        zip: 88888,
+        phone_number: "(555)555-5555",
+        description: "1.Person from Apply Index Test")
+      @application_2 = Apply.create!(name: "Fourth Person",
+        address: "96 There St",
+        city: "CityPlace",
+        state: "StateLocation",
+        zip: 88888,
+        phone_number: "(555)555-5555",
+        description: "2.Person from Apply Index Test")
+
+      PetApply.create!({pet: @pet_1, apply: @application_1})
+      PetApply.create!({pet: @pet_2, apply: @application_1})
+      PetApply.create!({pet: @pet_2, apply: @application_2})
+      PetApply.create!({pet: @pet_3, apply: @application_2})
+
+      @review_1 = @shelter_2.reviews.create!(title: 'Amazing', rating: 5, content: 'Found my best friend', image: 'https://live.staticflickr.com/7396/8728178651_912c2fa554_b.jpg')
+      @review_2 = @shelter_2.reviews.create!(title: 'Sucky', rating: 0, content: 'worst shelter ever. the manager yelled at me', image: 'https://www.peta.org/wp-content/uploads/2012/05/no-kill-gallery-03.jpg')
+      @review_3 = @shelter_1.reviews.create!(title: 'Yay', rating: 4, content: 'Loved it; the staff was so nice', image: 'https://www.clickorlando.com/resizer/neh8Wa4LQMzrQXwjJbpF89AP-Ms=/640x360/smart/filters:format(jpeg):strip_exif(true):strip_icc(true):no_upscale(true):quality(65)/arc-anglerfish-arc2-prod-gmg.s3.amazonaws.com/public/QV4N6JPXXVFCHP4QRTZYSEJEOY.jpg')
+      @review_4 = @shelter_1.reviews.create!(title: 'Best shelter eva!', rating: 5, content: 'I see how much they are doing to keep the animals happy', image: 'https://pyxis.nymag.com/v1/imgs/4bf/7d0/53374ff9319d2ebca19ebb532b1cec2b1d-17-shelter.rsquare.w1200.jpg')
+      @review_5 = @shelter_2.reviews.create!(title: 'Not the best', rating: 1, content: 'Not my favorite shelter, they only had pitbulls', image: 'https://www.citywatchla.com/images/stories/Sept-2017/075n.png')
+    end
+    it "can prevent pets from being deleted with pending applications" do
+      visit "/apply/#{@application_2.id}"
+      within "#application-pet-#{@pet_3.id}" do
+        check("Approve Application for this Pet")
+        click_button("Save changes")
+      end
+      visit "/pets/#{@pet_3.id}"
+
+      expect(page).to have_link("Delete Pet")
+      click_link "Delete Pet"
+      expect(page).to have_content("Unable to Delete a Pet with an approved application")
+      visit "/pets"
+      expect(page).to have_content(@pet_3.name)
+    end
+
+    it "it can delete a pet and it's associated favorited" do
+      visit "pets/#{@pet_4.id}"
+
+      click_button("Favorite This Pet!")
+      expect(page).to have_link("Delete Pet")
+      click_link "Delete Pet"
+      expect(current_path).to eq("/pets")
+      within"nav#nav-bar" do
+        expect(page).to have_link("My Favorites(0)")
```

Comment:
I like the way we checked the favorite was deleted on the page

Reply:
👍  i likes it